control_plane/attachment_service/src/http.rs

```diff
@@ -126,6 +202,63 @@ async fn handle_tenant_timeline_create(
     )
 }
 
+async fn handle_tenant_timeline_delete(
+    service: Arc<Service>,
+    req: Request<Body>,
+) -> Result<Response<Body>, ApiError> {
+    let tenant_id: TenantId = parse_request_param(&req, "tenant_id")?;
+    let timeline_id: TimelineId = parse_request_param(&req, "timeline_id")?;
+
+    deletion_wrapper(service, move |service| async move {
+        service.tenant_timeline_delete(tenant_id, timeline_id).await
+    })
+    .await
+}
+
+async fn handle_tenant_timeline_passthrough(
+    service: Arc<Service>,
+    req: Request<Body>,
+) -> Result<Response<Body>, ApiError> {
+    let tenant_id: TenantId = parse_request_param(&req, "tenant_id")?;
+
+    let Some(path) = req.uri().path_and_query() else {
+        // This should never happen, our request router only calls us if there is a path
+        return Err(ApiError::BadRequest(anyhow::anyhow!("Missing path")));
+    };
+
+    tracing::info!("Proxying request for tenant {} ({})", tenant_id, path);
+
+    // Find the node that holds shard zero
+    let (base_url, tenant_shard_id) = service.tenant_shard0_baseurl(tenant_id)?;
+
+    // Callers will always pass an unsharded tenant ID.  Before proxying, we must
+    // rewrite this to a shard-aware shard zero ID.
+    let path = format!("{}", path);
+    let tenant_str = tenant_id.to_string();
+    let tenant_shard_str = format!("{}", tenant_shard_id);
+    let path = path.replace(&tenant_str, &tenant_shard_str);
+
+    let client = mgmt_api::Client::new(base_url, service.get_config().jwt_token.as_deref());
+    let resp = client.proxy_get(path).await.map_err(|_e|
```

Comment:
>Also, not that the mgmt_api client still uses a lot of unwrap() and untyped anyhow errors;

Are you sure?  There aren't any unwraps or anyhow references in `mgmt_api.rs`

> Not sure how expensive the mgmt_api::Client is. It will probably not do any connection pooling; maybe we want that, maybe not.

They're pretty cheap: the expensive part is not re-using a TCP connection between requests.  It crossed my mind to do something more elaborate here, but in the near term, avoiding a connection pool is probably more robust than adding one that we don't exercise much.  If we add one, I'd want it to be available to everything that talks to a pageserver.  It might make sense to implement that at the same time as concurrency limits, so that we have a handle somewhere that controls all the concurrent calls to one pageserver.